src/egon/data/datasets/power_plants/pv_rooftop_buildings.py

```diff
@@ -0,0 +1,2433 @@
+
+    buildings_area_per_overlay_gdf = buildings_area_per_overlay_id(
+        valid_buildings_gdf,
+        grid_federal_state_gdf,
+    )
+
+    cap_per_bus_id_df = cap_per_bus_id(
+        buildings_area_per_overlay_gdf,
+        scenario_data(COMPONENT, CARRIER, scenario),
+    )
+
+    mastr_gdf = determine_end_of_life_gens(
+        mastr_gdf,
+        SCENARIO_TIMESTAMP,
+        PV_ROOFTOP_LIFETIME,
+    )
+
+    buildings_gdf = calculate_max_pv_cap_per_building(
+        valid_buildings_gdf,
+        mastr_gdf,
+        PV_CAP_PER_SQ_M,
+        ROOF_FACTOR,
+    )
+
+    mastr_gdf = calculate_building_load_factor(
+        mastr_gdf,
+        buildings_gdf,
+    )
+
+    probabilities_dict = probabilities(
+        mastr_gdf,
+        cap_ranges=CAP_RANGES,
+    )
+
+    cap_share_dict = cap_share_per_cap_range(
+        mastr_gdf,
+        cap_ranges=CAP_RANGES,
+    )
+
+    load_factor_dict = mean_load_factor_per_cap_range(
+        mastr_gdf,
+        cap_ranges=CAP_RANGES,
+    )
+
+    building_area_range_dict = building_area_range_per_cap_range(
+        mastr_gdf,
+        cap_ranges=CAP_RANGES,
+        min_building_size=MIN_BUILDING_SIZE,
+        upper_quantile=UPPER_QUNATILE,
+        lower_quantile=LOWER_QUANTILE,
+    )
+
+    allocated_buildings_gdf = desaggregate_pv(
+        buildings_gdf=buildings_gdf,
+        cap_df=cap_per_bus_id_df,
+        prob_dict=probabilities_dict,
+        cap_share_dict=cap_share_dict,
+        building_area_range_dict=building_area_range_dict,
+        load_factor_dict=load_factor_dict,
+        seed=SEED,
+        pv_cap_per_sq_m=PV_CAP_PER_SQ_M,
+    )
+
+    return frame_to_numeric(
+        add_buildings_meta_data(
+            allocated_buildings_gdf,
+            probabilities_dict,
+            SEED,
+        )
+    )
+
+
+class EgonPowerPlantPvRoofBuildingMapping(Base):
+    __tablename__ = "egon_power_plants_pv_roof_building_mapping"
+    __table_args__ = {"schema": "supply"}
+
+    einheit_mastr_nummer = Column(
+        String,
+        primary_key=True,
+        index=True,
+    )
+    building_id = Column(Integer, primary_key=True)
+
+
+def create_mapping_table(desagg_mastr_gdf):
+    """Create mapping table pv_unit <-> building"""
+    EgonPowerPlantPvRoofBuildingMapping.__table__.drop(
+        bind=engine, checkfirst=True
+    )
+    EgonPowerPlantPvRoofBuildingMapping.__table__.create(
+        bind=engine, checkfirst=True
+    )
+
+    desagg_mastr_gdf[["building_id"]].astype(int).reset_index().rename(
+        columns={"EinheitMastrNummer": "einheit_mastr_nummer"}
+    ).to_sql(
+        name=EgonPowerPlantPvRoofBuildingMapping.__table__.name,
+        schema=EgonPowerPlantPvRoofBuildingMapping.__table__.schema,
+        con=db.engine(),
+        if_exists="append",
+        index=False,
+    )
+
+
+class EgonPowerPlantPvRoofBuildingScenario(Base):
+    __tablename__ = "egon_power_plant_pv_roof_building_scenario"
```

Comment:
Please use `egon_power_plants_pv_roof_building` as table name (plants plural, remove suffix) to be in line with the other PP tables.

Also, please include a column `voltage_level` (cf. table `supply.egon_power_plants`):
- for existing units you can obtain it from MaStR (column "Spannungsebene", some example code [here](https://github.com/openego/eGon-data/blob/11e1616b083a5d087696d39774ea4594186186e5/src/egon/data/datasets/power_plants/__init__.py#L421-L432))
- for new units please set it acccording to our [definitions](https://redmine.iks.cs.ovgu.de/oe/projects/ego-n/wiki/Definition_of_thresholds_for_voltage_level_assignment)